src/rcc.rs

```diff
@@ -456,27 +503,73 @@ impl CFGR {
             while rcc.csr.read().lsirdy().bit_is_clear() {}
         }
 
-        // Turn on HSI48 if required
-        if self.hsi48 {
+        if self.msi != MsiFreq::NOMSI {
```

Comment:
Once the enum is changed, if `self.msi` is not `None` we can just write the enum as a u8 to the register